RealmSwift/RealmKeyedCollection.swift

```diff
@@ -0,0 +1,856 @@
+    override func setValue(_ value: Any?, forKey key: String) { base.setValue(value, forKey: key) }
+
+    // MARK: Filtering
+
+    override func filter(_ predicate: NSPredicate) -> Results<C.Value> { base.filter(predicate) }
+    override func contains(where predicate: @escaping (C.Key, C.Value) -> Bool) -> Bool { base.contains(where: predicate) }
+
+    // MARK: Sorting
+
+    override func sorted(ascending: Bool) -> Results<C.Value> { base.sorted(ascending: ascending) }
+    override func sorted(byKeyPath keyPath: String, ascending: Bool) -> Results<C.Value> { base.sorted(byKeyPath: keyPath, ascending: ascending) }
+    override func sorted<S>(by sortDescriptors: S) -> Results<C.Value> where S: Sequence, S.Element == SortDescriptor { base.sorted(by: sortDescriptors) }
+
+    // MARK: Indexes
+
+    override subscript(position: MapIndex) -> (key: C.Key, value: C.Value) { base[position] }
+    override func index(of object: C.Value) -> MapIndex? { base.index(of: object) }
+
+    // MARK: Aggregate Operations
+
+    override func min<T: MinMaxType>(ofProperty property: String) -> T? { base.min(ofProperty: property) }
+    override func max<T: MinMaxType>(ofProperty property: String) -> T? { base.max(ofProperty: property) }
+    override func sum<T: AddableType>(ofProperty property: String) -> T { base.sum(ofProperty: property) }
+    override func average<T: AddableType>(ofProperty property: String) -> T? { base.average(ofProperty: property) }
+
+    // MARK: Notifications
+
+    override func _observe(_ queue: DispatchQueue?, _ block: @escaping (RealmMapChange<Wrapper>) -> Void)
+        -> NotificationToken { base._observe(queue, block) }
+    override var isFrozen: Bool { base.isFrozen }
+
+    // MARK: Sequence Support
+
+    override func asNSFastEnumerator() -> Any {
+        return (base as! UntypedCollection).asNSFastEnumerator()
+    }
+    override func makeIterator() -> RLMMapIterator<SingleMapEntry<C.Key, C.Value>> {
+        base.makeIterator() as! RLMMapIterator<SingleMapEntry<C.Key, C.Value>>
+    }
+
+    // MARK: AssistedObjectiveCBridgeable
+
+    override class func bridging(from objectiveCValue: Any, with metadata: Any?) -> _AnyMap {
+        return _AnyMap(
+            base: (C.self as! AssistedObjectiveCBridgeable.Type).bridging(from: objectiveCValue, with: metadata) as! C)
+    }
+
+    override var bridged: (objectiveCValue: Any, metadata: Any?) {
+        return (base as! AssistedObjectiveCBridgeable).bridged
+    }
+
+
+    override func freeze() -> AnyMap<C.Key, C.Value> { return AnyMap(base.freeze()) }
+    override func thaw() -> AnyMap<C.Key, C.Value>? { return AnyMap(base.thaw()!)  }
+}
+/**
+ A type-erased `RealmKeyedCollection`.
+
+ Instances of `RealmKeyedCollection` forward operations to an opaque underlying collection having the same `Key`, `Value` type.
+ */
+public struct AnyMap<Key: _MapKey, Value: RealmCollectionValue>: RealmKeyedCollection {
```

Comment:
We only have one RealmKeyedCollection type so there doesn't appear to be anything this would be useful for?

Reply:
True and I'm all for code removal.